### PR TITLE
Add OpenBSD/arm64 support.

### PR DESCRIPTION
**Pull request by brad0**
Fix building on OpenBSD/arm64 and add CPU feature detection.

### CODE REVIEW

**Comment by AdmiralCurtiss** [comment]
Can someone who's familiar with ARM CPUs review this? @JosJuice maybe?

File: Source/Core/Common/ArmCPUDetect.cpp [hunk 1]
```diff
@@ -277,6 +282,33 @@ void CPUInfo::Detect()
...
+  if (sysctl(mib, 2, &isar0, &len, nullptr, 0) != -1)
+  {
+    if (ID_AA64ISAR0_AES(isar0) >= ID_AA64ISAR0_AES_PMULL)
+      bAES = true;
+    if (ID_AA64ISAR0_AES(isar0) >= ID_AA64ISAR0_SHA1_BASE)
+      bSHA1 = true;
+    if (ID_AA64ISAR0_AES(isar0) >= ID_AA64ISAR0_SHA2_BASE)
+      bSHA2 = true;
+    if (ID_AA64ISAR0_AES(isar0) >= ID_AA64ISAR0_CRC32_BASE)
+      bCRC32 = true;
```
Comment (JosJuice) [inline Source/Core/Common/ArmCPUDetect.cpp:310]: Shouldn't this be:

```suggestion
    if (ID_AA64ISAR0_AES(isar0) >= ID_AA64ISAR0_AES_BASE)
      bAES = true;
    if (ID_AA64ISAR0_SHA1(isar0) >= ID_AA64ISAR0_SHA1_BASE)
      bSHA1 = true;
    if (ID_AA64ISAR0_SHA2(isar0) >= ID_AA64ISAR0_SHA2_BASE)
      bSHA2 = true;
    if (ID_AA64ISAR0_CRC32(isar0) >= ID_AA64ISAR0_CRC32_BASE)
      bCRC32 = true;
```

**Review by JosJuice — APPROVED** [review]
I don't know OpenBSD, but it looks reasonable I guess.

File: Source/Core/Common/ArmCPUDetect.cpp (hunk 1 above)
Comment (brad0) [reply Source/Core/Common/ArmCPUDetect.cpp]: My bad. A bit too much copy and paste. I meant to correct the left side when I had initially adjusted the right side as appropriate. and with regard to AES_BASE vs AES_PMULL, that looks to be more appropriate for testing for just AES and not PMULL.

**Comment by brad0** [comment]
Thanks.

**Pull request merged**